src/main/scala/controllers/ApplicationController.scala

```diff
@@ -32,19 +33,51 @@ class ApplicationController @Inject()(actionHandler: ActionHandler, applications
   import ApplicationData._
   import FieldCheckHelpers._
 
-  def showSectionForm(id: ApplicationId, sectionNumber: Int) = Action.async { request =>
+  def editSectionForm(id: ApplicationId, sectionNumber: Int) = Action.async { request =>
     fieldsFor(sectionNumber) match {
-      case Some(fields) =>
-        applications.getSection(id, sectionNumber).flatMap { section =>
+      case Some(fields) => {
+        applications.getSection(id, sectionNumber).flatMap { section => {
           val hints = section.map(s => hinting(s.answers, checksFor(sectionNumber))).getOrElse(List())
           actionHandler.renderSectionForm(id, sectionNumber, section, questionsFor(sectionNumber), fields, noErrors, hints)
-        }
+        }}
+      }
+      // Do we want a generic error handler for these kind of unlikely eventualities (+ avoid compiler warnings)
+      case None => ???
```

Comment:
Need to fix this. Suggest returning a 404 response - fieldsFor now returns values for all the known sections, so a None would represent a non-existent section number, so NotFound seems reasonable.